### PR TITLE
Add the ability to validate modules in a namespace

### PR DESCRIPTION
**Pull request by binford2k**
This adds a few ways to specify modules to validate.

* It now defaults to the IAC supported module list instead of @eimlav's gist.
* It adds a --namespace argument to validate all modules in a namspace
* You can filter that search by supported/approved/partner
* Now supports YAML and JSON module lists both

It also cleans up a small bit of logic and corrects some incorrect
yarddoc comments. The specs could use some improvements.

This shouldn't change existing behaviour, but that was hard to validate
because I couldn't find examples to compare to. Everything I could infer
is roughly equivalent behaviour.

### CODE REVIEW

**Review by michaeltlombardi — APPROVED** [review]

**Comment by michaeltlombardi** [comment]
Left a little nit, but happy to merge if I'm wrong about it!

File: README.md
Comment (DavidS) [reply README.md:17]: yes it should have, and yes, you should have just gone in and fixed it and then hit the merge button

Comment (michaeltlombardi) [reply README.md]: Good point and well taken.

Comment (DavidS) [reply README.md]: :heart:

**Pull request merged**
